canonical-websites/build.snapcraft.io
# Update outdated npm modules

Fixes https://github.com/CanonicalLtd/snap-squad/issues/823

This PR updates most of dependencies to their latest minor (non breaking) version.
This helps to reduce number of issues found by `npm audit` (see https://github.com/CanonicalLtd/snap-squad/issues/823).

Some minor versions cannot be updated without breaking changes:
- `cookie-policy` - update breaks cookie-policy notification styles (BSI uses cookie-policy inside of React component, so it may need adjustment)
- `css-modules-require-hook` - update makes CSS modules fail when using `composes:` prop
- `vanilla-framework` - changes in styling and way BSI imports vanilla into CSS modules requires any update to Vanilla to be done with care and proper QA

### QA
- pull the branch
- Make sure clean install works properly:
- `rm -rf node_modules`
- `npm install`
- Run the site using the command `npm start -- --env=environments/dev.env`
- Check if everything works and looks as expected:
- Sign in, go to repo and build details, request builds, add repo, sign in to store to register name, remove repo...
- Make sure tests pass: `npm test`
- Make sure production build works: `npm build`
- Check if there are no unnecessary outdated minor versions: `npm outdated`

<img width="710" alt="Screenshot 2019-03-22 at 09 33 02" src="https://user-images.githubusercontent.com/83575/54812061-7eea6880-4c8a-11e9-947b-9cef21a3e56a.png">
